### PR TITLE
HBW-130 Make conditions disable_if and delete_if dynamic

### PR DESCRIPTION
**Pull request by mxsmirnoff**

### CODE REVIEW

**Comment by mxsmirnoff** [comment]
May-16-2019 18-15-31

File: hbw/app/javascript/packs/hbw/components/form/string.js.jsx [hunk 1]
```diff
@@ -12,12 +12,13 @@ modulejs.define('HBWFormString', ['React'], (React) => {
       const values = this.getNormalizedInitialValues(props.value || '');
 
       this.state = {
-        defaultValue:  values[0],
-        value:         values[1],
-        valid:         true,
-        visualValue:   values[2],
-        position:      values[3],
-        previousValue: null
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/form/string.js.jsx]: Not needed?

File: hbw/app/javascript/packs/hbw/components/helpers/conditions.js [hunk 2]
```diff
@@ -49,7 +49,9 @@ export default WrappedComponent => class WithConditions extends Component {
     }
   };
 
-  evaluateAndCondition = data => eval(data.condition.replace('$var', this.variableByName(data.variable)));
+  evaluateAndCondition = data => eval(data.condition.replace('$var', this.props.params.conditions_type === 'dynamic'
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/helpers/conditions.js]: Create separate function for retrieving value

File: hbw/app/javascript/packs/hbw/components/form/string.js.jsx (hunk 1 above)
Comment (mxsmirnoff) [reply hbw/app/javascript/packs/hbw/components/form/string.js.jsx]: Removed extra spaces

Comment (mxsmirnoff) [reply hbw/app/javascript/packs/hbw/components/form/string.js.jsx]: Oops

**Comment by mxsmirnoff** [comment]
Updated

**Comment by mxsmirnoff** [comment]
Updated

File: hbw/app/javascript/packs/hbw/components/form.js.jsx [hunk 3]
```diff
@@ -157,6 +161,10 @@ modulejs.define('HBWForm', ['React', 'jQuery', 'HBWError', 'HBWFormDatetime',
       return false;
     };
 
+    getChangedValues = ({ name, value }) => {
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/form.js.jsx]: getUpdatedFormValues

**Comment by mxsmirnoff** [comment]
Updated

File: hbw/app/javascript/packs/hbw/components/form/string.js.jsx [hunk 4]
```diff
@@ -270,6 +270,8 @@ modulejs.define('HBWFormString', ['React'], (React) => {
           visualValue: $el.value
         });
       }
+
+      this.props.trigger('hbw:update-value', { name: this.props.name, value: this.state.value });
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/form/string.js.jsx]: Let's make the function in the `WithCondition` HOC, something like `fireFieldValueUpdate(name, value)`.

File: hbw/app/javascript/packs/hbw/components/form.js.jsx [hunk 5]
```diff
@@ -157,6 +161,10 @@ modulejs.define('HBWForm', ['React', 'jQuery', 'HBWError', 'HBWFormDatetime',
       return false;
     };
 
+    getUpdatedFormValues = ({ name, value }) => {
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/form.js.jsx]: You call it `get...` but the only thing it does is `setState` that returns nothing.

**Comment by mxsmirnoff** [comment]
Updated

File: hbw/app/javascript/packs/hbw/components/form.js.jsx [hunk 6]
```diff
@@ -157,8 +161,24 @@ modulejs.define('HBWForm', ['React', 'jQuery', 'HBWError', 'HBWFormDatetime',
       return false;
     };
 
+    updateFormValues = ({ name, value }) => {
+      this.setState(prevState => ({ formValues: { ...prevState.formValues, [name]: value } }));
+    };
+
     notSerializableFields = () => ['static'];
 
+    getNameAndValueForm = () => {
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/form.js.jsx]: The main idea is that you get serializable fields. Isn't it?

**Comment by mxsmirnoff** [comment]
Updated

**Comment by mxsmirnoff** [comment]
Updated

File: hbw/app/javascript/packs/hbw/components/form.js.jsx [hunk 7]
```diff
@@ -157,8 +161,16 @@ modulejs.define('HBWForm', ['React', 'jQuery', 'HBWError', 'HBWFormDatetime',
...
     };
 
+    updateFormValues = ({ name, value }) => {
+      this.setState(prevState => ({ formValues: { ...prevState.formValues, [name]: value } }));
+    };
+
     notSerializableFields = () => ['static'];
 
+    setInitialValuesForm = () => {
+      this.setState({ formValues: this.props.variables });
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/form.js.jsx]: Can't we do it in the initial state definition?

**Comment by mxsmirnoff** [comment]
Updated

**Pull request merged**